### PR TITLE
Geotiff generation

### PR DESCRIPTION
**Pull request by spruceboy**
So this comes in two parts:

Part A uses Polar2Grid to make a bunch of "slices" out of the data, and makes the true color image. 
It requires a new(er) version of p2g. I tested with this version: 
http://hippy.gina.alaska.edu/distro/p2g_versions/polar2grid-swbundle-2.1.0rc6.tar.gz

Part B takes the output of part A, combines the slices into the various band combinations, and reformat them into nice clean geotiffs ready for gis clients.  It requires the processing-utils tools (the overview generation is used from that tool set) in the path, as well as a recent version of the mapping-tools builds. 

Part A needs a -m parameter with either viirs or modis
for example: 

> p2g_geotif.rb -m modis -t tmp in out

Part B needs a -m parameter with the platform - npp, a1, t1  .  This is needed instead of modis/viirs as aqua has a broken band 6, and so we are no longer making products with band 6 in it for aqua. 
for example: 

> feeder_geotif.rb -m t1 -t tmp in out

The images are going to look a bit different.  In particular, the scaling for the thermal images is completely different, and uses the same scaling as awips, so it is possible to map pixel values to brightness temperatures. 

Part A is also generating floating point tifs, with the "actual" data values in them.  This would be a good option for folks looking for brightness temperatures for example, or want more detail from the DNB images.   If they don't end up being useful to anyone, they can easily be turned off. 

MODIS is switch to using band 31 as the thermal band.  This might cause some issues for folks, but @cfdierking says this is a better option (and I agree).  

Let me know if any changes are needed, suggestions, etc.

### CODE REVIEW

File: bin/feeder_geotif.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+ENV['BUNDLE_GEMFILE'] = File.join(File.expand_path('../..', __FILE__), 'Gemfile')
+require 'bundler/setup'
+require 'fileutils'
+require_relative '../lib/processing_framework'
+
+class FeederGeotifClamp <  ProcessingFramework::CommandLineHelper
+  banner 'This tool takes MODIS and VIIRS data and makes geotifs intended for feeder.'
+  default_config 'feeder_geotif'
+
+  option ['-m', '--mode'], 'mode', 'The mode to use.', default: 'default'
```
Comment (teknofire) [inline bin/feeder_geotif.rb]: should this have a default?  It's going to try to pull out a set of configs that don't exist.  

Would probably be better if it was a required param and generated an error if it wasn't specified.

**Comment by teknofire** [comment]
looks like there is some sort of conflict with the bin/viirs_geotif on master

**Comment by spruceboy** [comment]
the viirs_geotif.rb thing needs to go away - I wasn't sure how to cleanly remove that.  It is replaced by this.

File: bin/feeder_geotif.rb (hunk 1 above)
Comment (spruceboy) [inline bin/feeder_geotif.rb]: That seems like a good plan - let me make that change.

**Comment by spruceboy** [comment]
@teknofire - I think I resolved the mode issue - it should be required with no default now. Let me know if you have any other question/sugestions etc.

**Pull request merged**